frontend/catalyst/jax_tracer.py

```diff
@@ -359,6 +356,33 @@ def trace_to_mlir(func, static_argnums, abstracted_axes, *args, **kwargs):
 
     # We remove implicit Jaxpr result values since we are compiling a top-level jaxpr program.
     jaxpr2, out_type2 = jaxpr_remove_implicit(jaxpr, out_type)
+
+    return jaxpr, jaxpr2, out_type2, out_tree
+
+
+def trace_to_mlir(func, static_argnums, abstracted_axes, *args, **kwargs):
+    """Lower a Python function into an MLIR module.
+
+    Args:
+        func: python function to be lowered
+        static_argnums: indices of static arguments.
+        abstracted_axes: abstracted axes specification. Necessary for JAX to use dynamic tensor
+            sizes.
+        args: arguments to ``func``
+        kwargs: keyword arguments to ``func``
+
+    Returns:
+        MLIR module: the MLIR module corresponding to ``func``
+        MLIR context: the MLIR context
+        ClosedJaxpr: the Jaxpr program corresponding to ``func``
+        jax.OutputType: Jaxpr output type (a list of abstract values paired with
+                        explicintess flags).
+        PyTreeDef: PyTree-shape of the return values in ``PyTreeDef``
+    """
+
+    jaxpr, jaxpr2, out_type2, out_tree = trace_to_jaxpr(
```

Comment:
`jaxpr2` ? 🤔

Reply:
This already exists in our code. I think it is a minor error that we return jaxpr instead of jaxpr2.

Reply:
Could we give it a different name? `jaxpr2` tells me very little about how it's different from `jaxpr`

Reply:
Same with the `2` for the `out_type` I guess.

Reply:
I think there's a bug here in our code. We always use `jaxpr` in our QJIT object but it should be `jaxpr2`. Looking above:

```
    # We remove implicit Jaxpr result values since we are compiling a top-level jaxpr program.
    jaxpr2, out_type2 = jaxpr_remove_implicit(jaxpr, out_type)
```

jaxpr2 is a postprocessed jaxpr. I want to leave this here until we actually fix the issue. This PR is already too big.

Reply:
Sure, can you open an issue with the bug? 

For the variable name I would still say there is no reason to propagate bad names to new code.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/477/commits/e44f48c7035857714fc7ca4ec1fdc025e85520e6